ams/server/src/main/resources/mysql/upgrade-0.4.1-to-0.5.0.sql

```diff
@@ -94,9 +94,9 @@ CREATE TABLE `table_runtime`
     `current_change_snapshotId`     bigint(20) DEFAULT NULL COMMENT 'Change table current snapshot id',
     `last_optimized_snapshotId`     bigint(20) NOT NULL DEFAULT '-1' COMMENT 'last optimized snapshot id',
     `last_optimized_change_snapshotId`     bigint(20) NOT NULL DEFAULT '-1' COMMENT 'last optimized change snapshot id',
-    `last_major_optimizing_time`    timestamp COMMENT 'Latest Major Optimize time for all partitions',
-    `last_minor_optimizing_time`    timestamp COMMENT 'Latest Minor Optimize time for all partitions',
-    `last_full_optimizing_time`     timestamp COMMENT 'Latest Full Optimize time for all partitions',
+    `last_major_optimizing_time`    datetime(3) DEFAULT NULL COMMENT 'Latest Major Optimize time for all partitions',
```

Comment:
What is the reason for changing the type to datetime here? Now that both timestamp and datetime are used in the table structure, what is our standard for using the two types?

Reply:
I think we use `datetime` only in order to make `DEFAULT NULL` valid, because `timestamp DEFAULT NULL` is illegal, and we will get an exception
```
mysql> create table test_table (create_time timestamp default null);
ERROR 1067 (42000): Invalid default value for 'create_time'
```

Reply:
As I know, the reason why you got the error is that field with timestamp type is not null by default, you will set the default null for timestamp type successfully by a statement like `create table test_table (create_time timestamp null default null);`

reference: read `[explicit_defaults_for_timestamp](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_explicit_defaults_for_timestamp)`

Reply:
You are right, it works! 

Maybe I should change all `datetime DEFAULT NULL` to `timestamp NULL DEFAULT NULL` to avoid the differences they dealing with time zones.

<img width="974" alt="image" src="https://github.com/NetEase/arctic/assets/103108928/b5940bb3-33cf-4ac5-a96c-25b5e68855c3">

Reply:
If there is no compelling reason to use `datetime`, I also recommend using a single type to store all timestamp information. Don't forget to also modify `uprade-0.4.1-to-0.5.0.sql`.